independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -1594,6 +1594,12 @@ static ResultHandle collectAnnotations(ClassOutput classOutput, ClassCreator bea
                         constructor.readStaticField(FieldDescriptor.of(InjectLiteral.class, "INSTANCE", InjectLiteral.class)));
             } else {
                 // Create annotation literal if needed
+
+                // we need to skip java lang annotations since the generated classes would end up in the java.lang package
+                // which the jvm does not allow
+                if (annotation.name().toString().startsWith("java.lang")) {
```

Comment:
I don't think we want to skip the literals but use a different package instead...

Reply:
What package do you propose?

Reply:
We should probably use `io.quarkus.arc.processor.AbstractGenerator.DEFAULT_PACKAGE` (this is already used for client proxies).

Reply:
Great thanks! I'll update the PR later on (or over the weekend).

Reply:
See also `io.quarkus.arc.processor.AnnotationLiteralGenerator.generatedSharedName()` and `io.quarkus.arc.processor.AnnotationLiteralGenerator.generatedLocalName()`.

Reply:
PR updated per your suggestion @mkouba. Thanks!